### ISSUE TITLE
abnormal increase in RAM

### ISSUE THREAD

**Issue by ehsansafari** [fewensa/telegram-client#27]
why, if I use this [line](https://github.com/fewensa/telegram-client/blob/7dfad7c7c7e901a4117f88353dcf92fbdc717cd5/examples/client_async.rs#L113) of code, the RAM memory will gradually increase with a lot of cash, so that 3 gigs will be added to the RAM memory in 5 minutes and 7 gigs will be added in 10 minutes.
`let chat = rasync_api.get_chat(GetChat::builder().chat_id(message.chat_id()).build()).await?;`

https://user-images.githubusercontent.com/19387708/132486745-31af8366-4427-4a7f-b0c7-c275bcd9e9bc.mp4

**but when I delete the** [line code](https://github.com/fewensa/telegram-client/blob/7dfad7c7c7e901a4117f88353dcf92fbdc717cd5/examples/client_async.rs#L113), **the RAM does not increase and remains constant?**

![2021-09-08 12_57_23-Window](https://user-images.githubusercontent.com/19387708/132486091-38eb8f30-804c-47c1-bbe9-217732876882.png)

source code : [telegram-client/examples/client_async.rs](https://github.com/fewensa/telegram-client/blob/master/examples/client_async.rs)
run code : cargo build --release

**Comment by fewensa** [comment]
I think the reason about there, https://github.com/fewensa/telegram-client/blob/7dfad7c7c7e901a4117f88353dcf92fbdc717cd5/src/rtd.rs#L39-L51, many threads are not dropped. I'm not sure, need to investigate this issue

**Comment by ehsansafari** [comment]
> , many threads are not dropped. I'm not sure, need to investigate this issue

exactly, i think all functions called from rasync_api() threads not dropped

**Comment by fewensa** [comment]
I'm not sure this is the reason now, I will investigate this issue, if you find anything, please let me know
